pygitguardian/client.py

```diff
@@ -676,6 +679,7 @@ def read_metadata(self) -> Optional[Detail]:
             "general__maximum_payload_size", MAXIMUM_PAYLOAD_SIZE
         )
         self.secret_scan_preferences = metadata.secret_scan_preferences
+        self.remediation_messages = metadata.remediation_messages
```

Comment:
I'm not sure to understand how this wont fail if the field does not exist in the endpoint's response.  
Also, we are missing tests IMO to test that the field is correctly created no matter the `/metadata` response. WDYT ?

Reply:
It does not fail because [the defaults to construct ServerMetadata are set there](https://github.com/GitGuardian/py-gitguardian/pull/111/files#diff-f729947e3a3359c7593744e8bd49edbbed0d8fee090eadb1d926541f5a55877fR758), but you are right there should be a test, I added it !